src/Illuminate/Database/PDO/MySqlDriver.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Illuminate\Database\PDO;
+
+use Doctrine\DBAL\Driver\AbstractMySQLDriver;
+use PDO;
+
+class MySqlDriver extends AbstractMySQLDriver
+{
+    public function connect(array $params)
```

Comment:
If accept my suggestion do not forget include Arr class.
```
    public function connect(array $params)
    {
        $pdoProperty = \Arr::get($params, 'pdo');
        if (! $pdoProperty  instanceof PDO) { // OR if (! ($pdoProperty = \Arr::get($params, 'pdo')) instanceof PDO) {
            throw new \InvalidArgumentException('Laravel requires the pdo property to be set and be a PDO instance.');
        }

        return new Connection($pdoProperty);
    }
```

With this suggestion we can avoid load `if conditions`, avoid call many times `$params['pdo']`.
In addition to, you should create a new Exception for this case, something like `InvalidArgumentPDOException` to avoid repeat 
` throw new \InvalidArgumentException('Laravel requires the pdo property to be set and be a PDO instance.');`
VS
`throw new \InvalidArgumentPDOException();`


On top of that, all the code inside the `connect` method is repeated multiple times, it's nice to encapsulate this method in a single responsibility method and use it however you need it.
helper, interface..

Reply:
> With this suggestion we can avoid load if conditions, avoid call many times $params['pdo'].

Unnecessary tbh.

> On top of that, all the code inside the connect method is repeated multiple times

True. I've just pushed a trait that encapsulates this and applied that to the three drivers that were repeating this piece of code. Thanks for the suggestion.